src/main/java/teammates/storage/api/StudentsDb.java

```diff
@@ -429,9 +428,8 @@ private void recreateStudentWithNewEmail(
         try {
             createStudent(new StudentAttributes(newCourseStudent), hasDocument);
         } catch (EntityAlreadyExistsException e) {
-            StudentAttributes existingStudent = (StudentAttributes) e.existingEntity;
-            String error = ERROR_UPDATE_EMAIL_ALREADY_USED
-                           + existingStudent.getName() + "/" + existingStudent.getEmail();
+            CourseStudent existingStudent = (CourseStudent) getEntity(new StudentAttributes(newCourseStudent));
```

Comment:
maybe refactor `new StudentAttributes(newCourseStudent)` out, since it's also needed in the try block. will make things clearer